www/pages/stop/stop-controller.js

```diff
@@ -117,16 +121,34 @@ angular.module('pvta.controllers').controller('StopController', function ($scope
             $scope.departuresByRoute.push(newDirsWithTimes);
           }
         });
-
-
         /* Step 4:
          * Download some details (name, color, etc) for each
          * route that has upcoming departures at this stop.
          */
         getRoutes(_.pluck($scope.departuresByRoute, 'RouteId'));
-      } // end highest if
+
+        /* Step 5:
+         * Sort the departures
+         * for each route.
+         */
+        var allSortedDepartures = [];
```

Comment:
Might be able to remove auxiliary array by reassigning departuresbyroute object's properties. Underscore fxn may be necessary